### PR TITLE
ubuntu_test: add shared utils for integration testing

### PR DESCRIPTION
**Pull request by jpnurmi**
Common integration test helpers split off from #420, to be shared with #283.

### CODE REVIEW

File: packages/ubuntu_test/lib/src/integration_test.dart [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+/// ".subiquity/etc/"
+final String configPath = p.join(subiquityPath, 'etc');
+
+/// e.g. ".subiquity/etc/wsl.conf"
+String configFile(String fileName) => p.join(configPath, p.basename(fileName));
+
+/// "integration_test/goldens/"
+final String goldenPath = p.join('integration_test', 'goldens');
+
+/// e.g. "integration_test/files/basic-setup/wsl.conf"
```
Comment (oSoMoN) [inline packages/ubuntu_test/lib/src/integration_test.dart:31]: The example doesn't look correct, it should be `"integration_test/goldens/wsl.conf"`.

File: packages/ubuntu_test/lib/src/integration_test.dart [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+/// Verifies that a subiquity config file matches a golden file.
+///
+/// NOTE: Use `flutter test --update-goldens` to update golden files.
+void verifyConfigFile(String fileName) {
+  _verifyGoldenFile(configFile(fileName), fileName);
+}
+
+void _verifyGoldenFile(String fileName, String goldenName) {
+  if (autoUpdateGoldenFiles) {
```
Comment (oSoMoN) [inline packages/ubuntu_test/lib/src/integration_test.dart:49]: I'm not seeing any other references to `autoUpdateGoldenFiles`, where does this come from?

File: packages/ubuntu_test/lib/src/integration_test.dart [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+Future<void> waitForSubiquityServer() async {
+  final startup = Completer();
+  callback(String socketPath) => startup.complete();
+
+  SubiquityServer.startupCallback = callback;
+  addTearDown(() => SubiquityServer.startupCallback = null);
+  return startup.future;
+}
+
+/// Cleans up the .subiquity direcrory to ensure a clean test environment for
```
Comment (oSoMoN) [inline packages/ubuntu_test/lib/src/integration_test.dart:70]: Typo: "directory"

**Review by oSoMoN — COMMENTED** [review]
A few minor comments, otherwise LGTM.

File: packages/ubuntu_test/lib/src/integration_test.dart (hunk 2 above)
Comment (jpnurmi) [reply packages/ubuntu_test/lib/src/integration_test.dart:50]: it's part of flutter_test and set to true when passing `--update-goldens` to `flutter test`. i added a comment with a link to the docs: https://api.flutter.dev/flutter/flutter_test/autoUpdateGoldenFiles.html

**Review by oSoMoN — APPROVED** [review]

**Pull request merged**